fortitude/src/rules/obsolescent/pause_statement.rs

```diff
@@ -0,0 +1,44 @@
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{AstRule, FromAstNode};
+use ruff_diagnostics::{Diagnostic, Fix, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_source_file::SourceFile;
+use tree_sitter::Node;
+
+/// ## What it does
+/// Checks for `pause` statements.
+///
+/// ## Why is this bad?
+/// `pause` statements were never properly standardised, doing different things
+/// on different compilers, and were completely removed in Fortran 95. They can
+/// usually be replaced with a simple call to `read(*,*)`
+///
+/// ## References
+/// - Metcalf, M., Reid, J. and Cohen, M., 2018, _Modern Fortran Explained:
+///   Incorporating Fortran 2018, Oxford University Press, Appendix B
+///   'Obsolescent and Deleted Features'
+#[violation]
+pub struct PauseStatement {}
+
+impl Violation for PauseStatement {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`pause` statements are a deleted feature")
+    }
+}
```

Comment:
Could we add a suggestion here?

```suggestion
    fn fix_title(&self) -> Option<String> {
        Some("Use 'read(*, *)' instead".into())
    }
}
```

Reply:
Oh yes, good point!